webpush/encrypt.go

```diff
@@ -109,7 +109,10 @@ func SubscriptionFromJSON(b []byte) (*Subscription, error) {
 			Auth   string
 		}
 	}
-	json.Unmarshal(b, &sub)
+	err := json.Unmarshal(b, &sub)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
could be written a bit shorter here:

    if err := json.Unmarshal(b, &sub); err != nil {
      return nil
    }